### ISSUE TITLE
Legacy Bloat

### ISSUE THREAD

**Issue by CraziFuzzy** [google/sagetv#68]
In the forum windows installer thread, Jeff did a great job of pulling some info from his memory on some of the binaries that are included in the v7 installer.  Many of which seemed to be pretty old an obsolete, and I'd like to compile a list of things to be marked for removal from v9, so not only can the binaries NOT be included, but references to such can be removed from the source.

Off the top of my head, the items I feel can be removed would be:
**Deinterlace.ax and all DI_*.dll's** - These are for the DScaler deinterlacer, which is pretty redundant with all of these deinterlacing methods built in to any modern decoder.  This would require not just removal of the binaries, but removal of the dscaler options in the STV, and likely code in the directshow library that builds the graphs.
**dlcllib.dll & tvtvDLCL.dll** - TVTV EPG related files - no longer used in v9, as TVTV was removed.
**stv*.ax & .dll** - The mainconcept (SageTV MPEG Decoder/Encoder/etc) codecs.  No license for use with the open source product, so can't be included.  Should probably bundle a different decoder - need to decide if ffmpeg or LAV should be included.  ffmpeg might work as a 'better' option than ac3filter (old version currently included) as well.

**Comment by stuckless** [comment]
In addition to this, I think it would be good to also define what is required for the Server and what is required for the Client.  This would likely help people if they are looking to create a 64bit Server, since looking at the all the dlls and .ax files that were posted, it's a bit overwhelming :)

**Comment by CraziFuzzy** [comment]
Yeah, my thoughts were to reduce what was obvious from it all, and then work out the server/client divide separately from a smaller list.  Unfortunately, the current architecture ties the server and it's standalone client together pretty tightly, and most the native libraries seem to be with native tuner access and playback.  Essentially, we'd need to make a server only 64-bit install, that would contain SageTVService64.exe and it's bare minimum support (for starters, throwing out native tuner support would make things simpler, especially if we rolled that native tuner support into a 32-bit network encoder - the new SageTVRecorder.exe).  I'm not even sure many capture devices include 64-bit compatible source filters.  I wonder, if instead of leveraging sagetv's internal tuner interface, if it might be better in the long run to just devote entirely to the network encoder interface, and maybe add general purpose encoders to OpenDCT for things like the HD-PVR and such.

**Comment by CraziFuzzy** [comment]
So next look.  **FreetypeFontJNI.dll**.  Is there really not a proper Java font rasterization that could be used in place of JNI calls?

**Comment by Narflex** [comment]
The TVTV stuff should already be completely removed. We removed it from the STV and all of the Java/JNI code.

DScaler I agree can be removed. Should be pretty easy to remove it from the STV, Java and native code. I can do that if nobody else wants to since it'll be easy for me. :)

The MainConcept decoders of course can't be in the open source installer...but ideally there'll be a way to have the installer leave those files there if they already exist so licensed V7 users upgrading get to keep those around. There may also be other assumptions in the code about their existence that I don't recall....but those assumptions should definitely be removed if they exist.

FreetypeFontJNI.dll should remain. There is of course proper Java font rasterization in the standard JRE...however, freetype was needed for a couple reasons:

1. Embedded platforms without AWT (this was the main reason)
2. Rendering consistency between server and client...Java didn't render the exact same size text between server and client and this led to layout issues in the placeshifter. However, since then, the code has been changed so the client just uses bitmaps for font drawing that the server creates...so this should no longer be an issue.
3. Freetype is much faster than the Java font rendering system as it can draw to native memory buffers whereas Java must draw to Java allocated memory (the native memory allows for faster transport across a network connection and less locking when accessing the memory natively).

The native image loading library should also really remain. It's far more reliable than using Java to do the image decoding; and it also supports on the fly decoding/scaling which greatly helps with decoding of very large images. It also has feature #3 above like Freetype (and #1 as well).

I was VERY happy when I finally had all the Java font & image stuff excised from SageTV and had it all using these native libraries instead. It greatly helped with performance and compatibility...these operations became far more predictable and consistent across systems and resolved pretty much all issues relating to these areas after it was done.  Neither of these should be much trouble to get to compile on Windows; as these both can be built using GNU tools...so these would likely be available to be cross compiled from a linux bot (whereas stuff using Microsoft libs like DirectShow related stuff may not be able to).

**Comment by jusjoken** [comment]
To further this discussion, from an installer perspective the following are no longer included in a Windows Install as of beta 0.003 of the installer...

- Deinterlace.ax and all the related DI_*. Note: no code changes or STV changes have been made yet.  The code compiles but if anything references these they likely will fail on run.

- the ms*.dll and related manifest were removed.  Note: a few of these were needed for the Main Concept Decoders so were handled by the routine in the installer that retains them before an uninstall of V7.

- PsisDecd.dll removed. Investigation found that on XP SP1 or earlier need this file

- stv*.* and legacy\stv*.* Main Concept decoders.  These are NOT part of a V9 install.  The installer has special handling to retain these IF they already exist on a system.

- AUDIO_PREVIEW.dll removed as the code that referenced this was previously removed from V9

- averapi_182.dll removed as the code that referenced this was previously removed from V9

- ttBdaDrvApi_Dll.dll and CIAP*.dll - Technotrend DVB card support files and were removed as support in the code had been removed from V9 previously

- tvtvDLCL.dll and dlcllib.dll removed from installer as TVTV support had been previously removed from V9

- FixP4.dll - Pentium 4 fix for the Provideo PVR256 card and ma32 and mase32 for the same card - removed as likely not in use by anyone (old)

- ksPropertyUtl.dll - Infrared receive support for the Provider PV256 card; removed however there may be references as pointed out by someone in SageTVInfraredReceive.cpp at line 55

**Comment by jusjoken** [comment]
From the remaining binaries at this time I have created a SupportBinaries archive that will likely be on BinTray to house the following that I have no current means to build from source but remain necessary for the windows install...

SageTV/Common

- as_ir200l.dll (Plugin for the Actisys IR200L IR blaster)
- pthreadGC2.dll (a GNU file for the pthread library on Windows - a FFMPEG dependency)
- pushreader.dll (native library for reading in push mode - third_party/SageTV-LGPL/Pushreader)
- uu_irsage.dll (USBUIRT library, source code is in third_party/USB-UIRT)

SageTV/SageTV
- avcodec-51.dll (FFMPEG library which I think is used by our StreamDemux)
- avformat-51.dll (FFMPEG library which I think is used by our StreamDemux)
- avutil-49.dll (FFMPEG library which I think is used by our StreamDemux)
- FreetypeFontJNI.dll (SageTV native library for platform independent text rendering (uses FreeType), the source code is in native/crosslibs/Freetype)
- iconv.dll (LGPL character conversion library)
- ImageLoader.dll (SageTV image loading library (platform independent), source code is in third_party/SageTV-LGPL/imageload)
- jhead.exe (used for manipulating EXIF tags on JPEG files)
- Mpeg2Transcoder.dll (SageTV library for doing transmuxing of MP3 files for playback on the MediaMVP. Source is in native/so/Mpeg2Transcoder)
- mvp.bin (Firmware for the MediaMVP)
- pthreadGC2.dll (a GNU file for the pthread library on Windows - a FFMPEG dependency)
- swscale.dll (SWScaler library (I thought this was compiled into the image loader DLL; this may also be another FFMPEG dependency from the Mpeg2Transcoder). Source code is in third_party/swscale)
